src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -208,12 +208,14 @@ private unsafe static PEReader OpenPEFile(string filePath, out MemoryMappedViewA
             // well for us since it can make IL access very slow (call to OS for each method IL query). We will map the file
             // ourselves to get the desired performance characteristics reliably.
 
+            FileStream fileStream = null;
             MemoryMappedFile mappedFile = null;
             MemoryMappedViewAccessor accessor = null;
-
             try
             {
-                mappedFile = MemoryMappedFile.CreateFromFile(filePath, FileMode.Open, null, 0, MemoryMappedFileAccess.Read);
+                fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
```

Comment:
There was an interesting scenario in cli CI when I submitted my changes to pass references from shared location. ILC opened a filestream from shared location which xUnit runner already had open while running the ILC E2E test. FileShare.None in CoreFX is too strict to allow another process to open. I was pulling my hair out as it always passed standalone.

I was surprised but suspected it was by design, so I posted an issue:
https://github.com/dotnet/corefx/issues/6092